### PR TITLE
Add self-upgrade step to update f-m package

### PR DESCRIPTION
**Pull request by Gauravtalreja1**
Description:
1. Add self-upgrade step to update f-m package
2. Combine satellite_capsule_upgrade for both y- and z-stream as similar to satellite-upgrade

Signed-off-by: Gaurav Talreja <gauravtalreja1@gmail.com>

### CODE REVIEW

File: upgrade/runner.py [hunk 1]
```diff
@@ -135,13 +133,10 @@ def product_upgrade_capsule(cap_host):
...
-                    execute(satellite_capsule_upgrade,
-                            cap_host, sat_host, host=cap_host)
-                elif settings.upgrade.from_version == settings.upgrade.to_version:
-                    execute(satellite_capsule_zstream_upgrade,
-                            cap_host, host=cap_host)
-                upgraded = execute(
-                    get_sat_cap_version, 'cap', host=cap_host)[cap_host]
+                    execute(satellite_capsule_upgrade, cap_host, sat_host, host=cap_host)
+                else:
+                    execute(satellite_capsule_upgrade, cap_host, True, host=cap_host)
```
Comment (lpramuk) [inline upgrade/runner.py:138]: In else clause `sat_host = True` ?  I don't think this is correct

**Review by lpramuk — CHANGES_REQUESTED** [review]
Parameters mismatch

File: upgrade/runner.py (hunk 1 above)
Comment (Gauravtalreja1) [reply upgrade/runner.py]: @lpramuk Good catch! updated, request to re-review.
Thank you :)

File: upgrade/runner.py [hunk 2]
```diff
@@ -135,13 +133,10 @@ def product_upgrade_capsule(cap_host):
...
-                    execute(satellite_capsule_upgrade,
-                            cap_host, sat_host, host=cap_host)
-                elif settings.upgrade.from_version == settings.upgrade.to_version:
-                    execute(satellite_capsule_zstream_upgrade,
-                            cap_host, host=cap_host)
-                upgraded = execute(
-                    get_sat_cap_version, 'cap', host=cap_host)[cap_host]
+                    execute(satellite_capsule_upgrade, cap_host, sat_host, host=cap_host)
+                else:
+                    execute(satellite_capsule_upgrade, cap_host, sat_host, True, host=cap_host)
```
Comment (lpramuk) [inline upgrade/runner.py:138]: Isn't it better to have just this:
```
zstream = settings.upgrade.from_version == settings.upgrade.to_version
execute(satellite_capsule_upgrade, cap_host, sat_host, zstream, host=cap_host)
```

**Review by lpramuk — CHANGES_REQUESTED** [review]
Please simplify the code a bit

File: upgrade/capsule.py [hunk 3]
```diff
@@ -83,35 +83,41 @@ def satellite_capsule_upgrade(cap_host, sat_host):
...
+    ]
     with fabric_settings(warn_only=True):
-        if os_ver == 7:
-            enable_disable_repo(enable_repos_name=[RH_CONTENT['ansible']['label']])
-        if settings.upgrade.distribution == 'cdn':
-            enable_disable_repo(enable_repos_name=[capsule_repo, maintenance_repo, client_repo])
+        if settings.upgrade.distribution == "cdn":
+            enable_disable_repo(enable_repos_name=capsule_repos)
+            if os_ver == 7:
+                enable_disable_repo(enable_repos_name=[RH_CONTENT['ansible']['label']])
```
Comment (lpramuk) [inline upgrade/capsule.py:114]: Wrong place - we need ansible from CDN always !
Put away from `if settings.upgrade.distribution == "cdn"` as it was originally

**Review by lpramuk — CHANGES_REQUESTED** [review]
Ansible repo enablement misplace

File: upgrade/capsule.py (hunk 3 above)
Comment (Gauravtalreja1) [reply upgrade/capsule.py]: Updated

**Review by lpramuk — APPROVED** [review]

**Comment by Gauravtalreja1** [comment]
@lpramuk Thank you! ❤️

**Pull request merged**